### PR TITLE
add zeronet

### PR DESCRIPTION
**Pull request by specter119**
#### Adding a new cask
- [x] Checked there aren’t open [pull requests](https://github.com/caskroom/homebrew-cask/pulls) for the same cask.
- [x] Checked there aren’t closed [issues](https://github.com/caskroom/homebrew-cask/issues) where that cask was already refused.
- [x] When naming the cask, followed the [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md).
- [x] Commit message includes cask’s name.
- [x] `brew cask audit --download {{cask_file}}` is error-free.
- [x] `brew cask style --fix {{cask_file}}` left no offenses.
- [x] `brew cask install {{cask_file}}` worked successfully.
- [x] `brew cask uninstall {{cask_file}}` worked successfully.

### CODE REVIEW

File: Casks/zeronet.rb [hunk 1]
```diff
@@ -0,0 +1,11 @@
+cask 'zeronet' do
+  version :latest
+  sha256 'aab07d27cd8f3fd307a3438a953af3bdfc7de62592c375a9463028ab7c7331e3'
```
Comment (pnicolai) [inline Casks/zeronet.rb]: with `version :latest` you should use `:no_check`

File: Casks/zeronet.rb [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+  url 'https://github.com/HelloZeroNet/ZeroBundle/raw/master/dist/ZeroBundle-mac-osx.zip'
```
Comment (vitorgalvao) [inline Casks/zeronet.rb]: Download instead from the [releases page](https://github.com/HelloZeroNet/ZeroBundle/releases).

File: Casks/zeronet.rb [hunk 3]
```diff
@@ -0,0 +1,11 @@
+cask 'zeronet' do
+  version :latest
```
Comment (vitorgalvao) [inline Casks/zeronet.rb]: After you update `url`, this can be versioned.

File: Casks/zeronet.rb (hunk 2 above)
Comment (vitorgalvao) [inline Casks/zeronet.rb]: Please add an [`appcast`](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md).

Comment (vitorgalvao) [inline Casks/zeronet.rb]: Needs a `# github.com/HelloZeroNet/ZeroBundle was verified as official when first introduced to the cask` comment above this line ([example](https://github.com/caskroom/homebrew-cask/blob/4ad02c9f087deb9e553ab0129c079d73c474add8/Casks/0ad.rb#L5); [reasoning](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/url.md#when-url-and-homepage-hostnames-differ-add-a-comment)).

Comment (specter119) [inline Casks/zeronet.rb]: @vitorgalvao releases version is unable to run in my machine, and the master version is the official link.

Comment (vitorgalvao) [inline Casks/zeronet.rb]: They are both official. But one of them is versioned, and we prefer that.

Comment (vitorgalvao) [inline Casks/zeronet.rb]: I see the releases are outdated.

All the other points stand, though.

Comment (specter119) [inline Casks/zeronet.rb]: so there is no need for [appcast](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/stanzas/appcast.md)?

Comment (vitorgalvao) [inline Casks/zeronet.rb]: There is. As soon as they update the releases page, we can update the cask as well.

File: Casks/zeronet.rb (hunk 3 above)
Comment (vitorgalvao) [inline Casks/zeronet.rb]: Set the version to the latest on in the releases page.

Comment (vitorgalvao) [inline Casks/zeronet.rb]: Like I said, set the version to the latest on in the releases page.

File: Casks/zeronet.rb [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+  sha256 :no_check
+
+  # zeronet.io and https://github.com/HelloZeroNet/ZeroBundle were verified as official when first introduced to the cask
+  url 'https://github.com/HelloZeroNet/ZeroBundle/raw/master/dist/ZeroBundle-mac-osx.zip'
```
Comment (vitorgalvao) [inline Casks/zeronet.rb]: Still missing the appcast.

File: Casks/zeronet.rb [hunk 5]
```diff
@@ -0,0 +1,12 @@
+cask 'zeronet' do
+  version :latest
+  sha256 :no_check
+
+  # zeronet.io and https://github.com/HelloZeroNet/ZeroBundle were verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/zeronet.rb]: The comment needs to be exactly like I told you. It’s standard.

File: Casks/zeronet.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+cask 'zeronet' do
```
Comment (vitorgalvao) [inline Casks/zeronet.rb]: You checked the box, but the token reference was not followed.

This has to be changed to `cask 'zerobundle' do`. The [token reference](https://github.com/caskroom/homebrew-cask/blob/master/doc/cask_language_reference/token_reference.md) must be followed, to avoid duplicates.

The file’s name also needs to be changed to `zerobundle.rb`.

File: Casks/zeronet.rb [hunk 7]
```diff
@@ -0,0 +1,12 @@
...
+  version :latest
+  sha256 :no_check
+
+  # zeronet.io and https://github.com/HelloZeroNet/ZeroBundle were verified as official when first introduced to the cask
+  url 'https://github.com/HelloZeroNet/ZeroBundle/raw/master/dist/ZeroBundle-mac-osx.zip'
+  name 'zeronet'
```
Comment (vitorgalvao) [inline Casks/zeronet.rb]: `name 'ZeroNet ZeroBundle'`.

**Comment by specter119** [comment]
@victorpopkov is it right?

File: Casks/zerobundle.rb [hunk 8]
```diff
@@ -0,0 +1,13 @@
+cask 'zerobundle' do
+  version '0.1.1'
+  sha256 'a8bffd28471aae1fc70098700af9cd84561a0d0f2e053b32504ae0f06388a47e'
+
+  # zeronet.io and https://github.com/HelloZeroNet/ZeroBundle were verified as official when first introduced to the cask
```
Comment (vitorgalvao) [inline Casks/zerobundle.rb]: Comment is still not correct. Needs to be `# github.com/HelloZeroNet/ZeroBundle was verified as official when first introduced to the cask`.

File: Casks/zerobundle.rb [hunk 9]
```diff
@@ -0,0 +1,13 @@
...
+  url "https://github.com/HelloZeroNet/ZeroBundle/releases/download/#{version}/ZeroBundle-v#{version}.zip"
+  appcast 'https://github.com/HelloZeroNet/ZeroBundle/releases.atom'
```
Comment (vitorgalvao) [inline Casks/zerobundle.rb]: `appcast` needs a `checkpoint:`.

File: Casks/zerobundle.rb [hunk 10]
```diff
@@ -0,0 +1,13 @@
...
+  name 'ZeroBundle'
+  homepage 'https://zeronet.io/'
+  license :gpl # TODO: change license and remove this comment; ':unknown' is a machine-generated placeholder
+
+  suite 'ZeroNet'
```
Comment (vitorgalvao) [inline Casks/zerobundle.rb]: Why did you change this? Does this need to be the whole directory? If so, the cask token is now wrong again.

File: Casks/zerobundle.rb [hunk 11]
```diff
@@ -0,0 +1,13 @@
+cask 'zerobundle' do
+  version '0.1.1'
+  sha256 'a8bffd28471aae1fc70098700af9cd84561a0d0f2e053b32504ae0f06388a47e'
+
+  # zeronet.io and https://github.com/HelloZeroNet/ZeroBundle were verified as official when first introduced to the cask
+  url "https://github.com/HelloZeroNet/ZeroBundle/releases/download/#{version}/ZeroBundle-v#{version}.zip"
+  appcast 'https://github.com/HelloZeroNet/ZeroBundle/releases.atom'
+  name 'ZeroBundle'
+  homepage 'https://zeronet.io/'
+  license :gpl # TODO: change license and remove this comment; ':unknown' is a machine-generated placeholder
```
Comment (vitorgalvao) [inline Casks/zerobundle.rb]: Remove the comment (see what it says).

**Comment by vitorgalvao** [comment]
@specter119 Are you still interested in getting this merged?

File: Casks/zerobundle.rb (hunk 11 above)
Comment (specter119) [inline Casks/zerobundle.rb]: @vitorgalvao yes, but i have some trouble in understanding ' checkpoint'.

Comment (vitorgalvao) [inline Casks/zerobundle.rb]: Do the other changes and answer the other questions, and we’ll take care of that in the end.

File: Casks/zerobundle.rb (hunk 10 above)
Comment (specter119) [inline Casks/zerobundle.rb]: I am really sorry for makes you troubles. I made a mass, so i thought i should make it clear, then i didn't answer your question.

This is because i have delete the branch by mistake. it really need the whole folder. the cask is wrong because i took **the wrong zip url** and also get the **wrong sha256**.

The right parts should be:
```
    sha256 '1ba5a80b5e95ab62edbd435c1dc89bbdf8f215945f887681e5f54115984cb048'
    url "https://github.com/HelloZeroNet/ZeroBundle/releases/download/#{version}/ZeroBundle-mac-v#{version}.zip"
```
What's more, this app have at least 2 python package dependence: msgpack-python and gevent. 

That's all thank you!

Comment (vitorgalvao) [inline Casks/zerobundle.rb]: We can’t satisfy python dependencies, so it doesn’t really makes sense to add this as a cask.

Since you had the troubles with the branch and cannot revisit this for the next few days, lets try again then. This is more complex than what I had expected, to make the fixes for you, since I do not know the software you’re adding.

Please submit the new PR when you can again (or perhaps someone else will).

Comment (specter119) [inline Casks/zerobundle.rb]: OK thank you.
